conf/modules.config

```diff
@@ -447,24 +447,97 @@ process {
     }
 
     withName: ".*DEDUPLICATE:SAMTOOLS_INDEX" {
-        ext.prefix = { "${meta.id}_${meta.library_id}_dedupped" }
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_dedupped" }
         ext.args = { params.fasta_largeref ? "-c" : "" }
         publishDir = [
             path: { "${params.outdir}/deduplication/" },
             mode: params.publish_dir_mode,
-            pattern: '*.bai'
+            pattern: '*.{bai,csi}'
         ]
     }
 
     withName: ".*DEDUPLICATE:SAMTOOLS_FLAGSTAT" {
-        ext.prefix = { "${meta.id}_${meta.library_id}_dedupped" }
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_dedupped" }
         publishDir = [
             path: { "${params.outdir}/deduplication/" },
             mode: params.publish_dir_mode,
             pattern: '*.flagstat'
         ]
     }
 
+    withName: MAPDAMAGE2 {
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_rescaled" }
+        ext.args = { [
+            "--rescale",
+            // "--rescale-out=${task.ext.prefix}.bam", // This doesn't work, because the output is expected to be in the mapdamage results dir.
+            "--seq-length=${params.damage_manipulation_rescale_seqlength}",
+            params.damage_manipulation_rescale_length_3p != 0 ? "--rescale-length-3p=${params.damage_manipulation_rescale_length_3p}" : "",
+            params.damage_manipulation_rescale_length_5p != 0 ? "--rescale-length-5p=${params.damage_manipulation_rescale_length_5p}" : "",
+            { meta.strandedness } == "single" ? '--single-stranded' : ''
+        ].join(' ').trim() }
+        publishDir = [
+            path: { "${params.outdir}/damage_manipulation/" },
+            mode: params.publish_dir_mode,
+            pattern: 'result*/*.rescaled.bam',
+            saveAs: { "${meta.sample_id}_${meta.library_id}_rescaled.bam" } // Needed to save the bam directly in the output dir instead of within the mapdamage results dir.
+        ]
+    }
+
+    withName: SAMTOOLS_INDEX_DAMAGE_RESCALED {
+        ext.args = { params.fasta_largeref ? "-c" : "" }
+        publishDir = [
+            path: { "${params.outdir}/damage_manipulation/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.{bai,csi}',
+            saveAs: { "${meta.sample_id}_${meta.library_id}_rescaled.bam.bai" } // ext.prefix cannot be used here, so rename for publishing instead.
```

Comment:
Again why not?

Reply:
The rescaled bam is always named `rescaled.bam` by the mapDamage module. This gets passed on to SAMTOOLS_INDEX with that name and hence the index is always called `rescaled.bam.bai`. This way the index gets an appropriate name matching the rescaled bam output in the publish dir